lib/va_profile/profile_information/service.rb

```diff
@@ -0,0 +1,201 @@
+      def create_or_update_info(http_verb, record)
+        with_monitoring do
+          icn_with_aaid_present!
+          model = record.class
+          http_verb = http_verb.to_sym == :update ? reassign_http_verb(record) : http_verb.to_sym
+          update_path = record.try(:permission_type).present? ? 'permissions' : record.contact_info_attr.pluralize
+          raw_response = perform(http_verb, update_path, record.in_json)
+          response = model.transaction_response_class.from(raw_response)
+          return response unless http_verb == :put && record.contact_info_attr == 'email' && old_email.present?
+
+          transaction = response.transaction
+          return response unless transaction.received?
+
+          # Create OldEmail to send notification to user's previous email
+          OldEmail.create(transaction_id: transaction.id, email: old_email)
+        end
+      rescue => e
+        handle_error(e)
+      end
+
+      # def get_transaction_status(transaction_id, model)
+      #   with_monitoring do
+      #     icn_with_aaid_present!
+      #     raw_response = perform(:post, model.transaction_status_path(@user, transaction_id), '')
+      #     VAProfile::Stats.increment_transaction_results(raw_response)
+      #     transaction_status = model.transaction_response_class.from(raw_response, @user)
+      #     return transaction_status unless model.send_change_notifcations?
+
+      #     send_change_notifications(transaction_status)
+      #     transaction_status
+      #   end
+      # rescue => e
+      #   handle_error(e)
+      # end
+
+      private
+
+      def icn_with_aaid
+        return "#{@user.idme_uuid}^PN^200VIDM^USDVA" if @user.idme_uuid
+        return "#{@user.logingov_uuid}^PN^200VLGN^USDVA" if @user.logingov_uuid
+
+        nil
+      end
+
+      def icn_with_aaid_present!
+        raise 'User does not have a icn' if icn_with_aaid.blank?
+      end
+
+      def vet360_id_present!
+        raise 'User does not have a vet360_id' if @user&.vet360_id.blank?
+      end
+
+      def path
+        "#{OID}/#{ERB::Util.url_encode(icn_with_aaid)}"
+      end
+
+      def old_email(transaction_id: nil)
+        return @user.va_profile_email if transaction_id.nil?
+
+        OldEmail.find(transaction_id).try(:email)
+      end
+
+      # create_or_update cannot determine if record exists
+      # Reassign :update to either :put or :post
+      def reassign_http_verb(record)
+        contact_info = VAProfileRedis::ContactInformation.for_user(@user)
+        attr = record.contact_info_attr
+        raise "invalid #{record.model} VAProfile::ProfileInformation" if attr.nil?
+
+        record.id = contact_info.public_send(attr)&.id
+        record.id.present? ? :put : :post
+      end
+
+      def get_email_personalisation(type)
+        { 'contact_info' => EMAIL_PERSONALISATIONS[type] }
+      end
+
+      # def send_change_notifications(transaction_status)
```

Comment:
Is there a reason the following methods are commented out? Looks like it hooks into VANotify to send notifications

Reply:
These will be uncommented in other PRs :)